File: modules/project/src/main/java/com/falsepattern/zigbrains/project/steps/ui/BaseNodeDescriptor.java
package com.falsepattern.zigbrains.project.steps.ui;

import com.intellij.ide.projectView.PresentationData;
import com.intellij.ide.util.treeView.PresentableNodeDescriptor;
import com.intellij.openapi.project.Project;
import com.intellij.ui.SimpleTextAttributes;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.Icon;

public class BaseNodeDescriptor<T> extends PresentableNodeDescriptor<T> {
    private String description;
    public BaseNodeDescriptor(@Nullable Project project, String displayName, Icon displayIcon) {
        this(project, displayName, null, displayIcon);
    }
    public BaseNodeDescriptor(@Nullable Project project, String displayName, String description, Icon displayIcon) {
        super(project, null);
        setIcon(displayIcon);
        myName = displayName;
        this.description = description;
        update();
    }

    @Override
    protected void update(@NotNull PresentationData presentation) {
        presentation.setIcon(getIcon());
        presentation.addText(myName, SimpleTextAttributes.REGULAR_ATTRIBUTES);
        presentation.setTooltip(description);
    }

    @Override
    public T getElement() {
        return null;
    }
}

File: core/src/main/kotlin/com/falsepattern/zigbrains/project/steps/ui/BaseNodeDescriptor.kt
/*
 * This file is part of ZigBrains.
 *
 * Copyright (C) 2023-2025 FalsePattern
 * All Rights Reserved
 *
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 *
 * ZigBrains is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, only version 3 of the License.
 *
 * ZigBrains is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with ZigBrains. If not, see <https://www.gnu.org/licenses/>.
 */

package com.falsepattern.zigbrains.project.steps.ui

import com.intellij.ide.projectView.PresentationData
import com.intellij.ide.util.treeView.PresentableNodeDescriptor
import com.intellij.openapi.project.Project
import com.intellij.ui.SimpleTextAttributes
import javax.swing.Icon

open class BaseNodeDescriptor<T>(project: Project?, displayName: String, displayIcon: Icon? = null, private var description: String? = null): PresentableNodeDescriptor<T>(project, null) {
    init {
        icon = displayIcon
        myName = displayName
        update()
    }

    override fun update(presentation: PresentationData) {
        presentation.setIcon(icon)
        presentation.addText(myName, SimpleTextAttributes.REGULAR_ATTRIBUTES)
        presentation.tooltip = description
    }

    override fun getElement(): T? {
        return null
    }
}